lib/puppet/type/acl.rb

```diff
@@ -226,6 +226,9 @@ def initialize(*args)
          to the manifest.
          Reference: #{value.inspect}"
       end
+      if !(value['type'].nil? || value['type'].empty?)
+        warning('Permission `type` is deprecated and has been replaced with perm_type for allow or deny')
+      end
```

Comment:
We already have a precedent of validating the ace values within the ace. https://github.com/puppetlabs/puppetlabs-acl/blob/master/lib/puppet/type/acl/ace.rb#L171 | https://github.com/puppetlabs/puppetlabs-acl/blob/master/lib/puppet/type/acl/ace.rb#L151 | https://github.com/puppetlabs/puppetlabs-acl/blob/master/lib/puppet/type/acl/ace.rb#L191

Is this different enough to warrant validation in acl.rb?